app/src/main/kotlin/com/simplemobiletools/musicplayer/models/Track.kt

```diff
@@ -52,6 +53,9 @@ data class Track(
             }
             sorting and PLAYER_SORT_BY_TRACK_ID != 0 -> {
                 when {
+                    discNumber <= 0 && other.discNumber > 0 -> 1
+                    discNumber > 0 && other.discNumber <= 0 -> -1
+                    discNumber != other.discNumber -> AlphanumericComparator().compare(discNumber.toString(), other.discNumber.toString())
```

Comment:
no need to use alphanumeric comparator if the value is an int, it is needed at strings only. Guess we dont need it a couple lines lower at "else" with comparing trackId either.